### DISCUSSION TITLE
Scroll position & wire:navigate

### DISCUSSION THREAD

**Discussion by martin-ro** [livewire/livewire#6220] [1. Help] [answered]
So wire:navigate is pretty great! The only thing is it keeps the scroll position. Navigating between longer content pages is kinda weird.

To get around that I use:
```js
window.addEventListener('livewire:navigated', () => {
    window.scrollTo({ top: 0 });
});
```

Works great!

Now when I list records and use `$records->links()` to paginate I have the scroll position problem again. To fix that I created a trait `WithPaginationAndScroll.php`:

```php
trait WithPaginationAndScroll
{
    use WithPagination {
        WithPagination::gotoPage as parentGotoPage;
        WithPagination::nextPage as parentNextPage;
        WithPagination::previousPage as parentPreviousPage;
    }

    public function gotoPage($page): void
    {
        $this->parentGotoPage($page);

        $this->dispatch('paginator-updated');
    }

    public function nextPage(): void
    {
        $this->parentNextPage();

        $this->dispatch('paginator-updated');
    }

    public function previousPage(): void
    {
        $this->parentPreviousPage();

        $this->dispatch('paginator-updated');
    }
}
```

and use:
```js
Livewire.on("paginator-updated", function () {
    window.scrollTo({ top: 0, behavior: "smooth" });
});
```

This is kind of nice when paginating since the smooth scroll indicates I have paginated (useful when records have very similar UI). So good so far... only thing is it randomly fails:

click next link: scrolls smooth to top.
click next link: scroll position stays the same.
click next link: scrolls smooth to top.
click next link: scrolls smooth to top.
click next link: scroll position stays the same.
and so on...

Has anyone an idea what's going wrong or maybe any better solution?

**Comment by lrljoe** [comment]
My suggestion? Publish the pagination blades, and add your window.scrollTo in there, so that you guarantee it fires smoothly.
If you combine that with the loading icon then it should look pretty slick.

**Comment by martin-ro** [comment]
I already tried adding
```js
<script>
    Livewire.on("paginator-updated", function () {
        window.scrollTo({ top: 0, behavior: "smooth" });
    });
</script>
```

to the published paginator. Same result... 

That's what you meant right?

**Comment by joshhanley** [comment]
@martin-ro so as it turns out navigate does actually scroll to the top of page. Is this not happening for you? See https://github.com/livewire/livewire/commit/5ec6a38965dd5233e6295c418ab1c689314c956c

Not sure you are aware, but the paginator doesn't use navigate, hence the scroll isn't handled, so you will need to handle that yourself. Rather than overriding all the nextPage/previousPage methods, you can use the `updatedPaginator()` hook to dispatch your event, see https://livewire.laravel.com/docs/pagination#general-paginator-hooks

Hope this helps!

**Reply by martin-ro** [reply]
@joshhanley Paginator aside for a moment, navigate does not scroll for me. I have set up a fresh install that demonstrates that the scroll position doesn't change: https://github.com/martin-ro/lw3-scroll

If you scroll down halfway and click one of the user names, you will get halfway scrolled down on that page.

**Reply by martin-ro** [reply]
Just a quick update: It looks like when I refresh the page the first click is scrolled to the top. All subsequent clicks are not

**Comment by orkuncoding** [comment]
Hello Martin, did you solve the problem ? I have same issue too.

**Comment by joshhanley** [comment, marked as answer]
We're in the process of investigating all wire:navigate issues and I have tested out your scenario. I believe this has been fixed so I'm going to close this. If you're still experiencing problems, please open a new discussion. Thanks!
